### ISSUE TITLE
JSON allowComments not working

### ISSUE THREAD

**Issue by Here21** [microsoft/monaco-editor#2426]
![image](https://user-images.githubusercontent.com/12051024/113867267-e8416380-97e0-11eb-8a81-1a6199a8fc50.png)

Hi, I upgrade 0.21 -> 0.23, and then I find my JSON Editor has something strange that is different from version 0.21.

For example:
![image](https://user-images.githubusercontent.com/12051024/113868122-e7f59800-97e1-11eb-92c3-3583d748ce0c.png)
it's not allow comments and monaco-editor will throw error with "Trailing comma" warning like below.
but 0.23 not do that.

my settings:
```js
 monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
      validate: true,
      allowComments: false,
      schemaValidation: 'error'
    });
```

So it's a bug?

**Comment by gluck** [comment]
Getting this behavior as well, `allowComments` no longer works since change https://github.com/microsoft/monaco-json/commit/22746ae61a9191a3eedfd89ae49aeef32f922adf
Passing json diagnostics options as document language settings like this is probably not intended ?
In particular `this.commentSeverity` (set from `allowComments`) is [no longer used](https://github.com/microsoft/vscode-json-languageservice/blob/main/src/services/jsonValidation.ts#L53):

```
let commentSeverity = documentSettings ? toDiagnosticSeverity(documentSettings.comments) : this.commentSeverity;
```

Although it opens new (undocumented/untyped) capabilities for json diagnostics options, such as:
```
{
  "comments": "error" // <-- same as allowComments = false, but working
  "trailingCommas": "error" // <-- same as previous behavior, but now configurable
}
```

**Comment by aeschli** [comment]
It's a bug in the json-languageservice. Fixed https://github.com/microsoft/vscode-json-languageservice/commit/d3160a7ac3fda283a4a4964be36f70b6b7627109

**Comment by Here21** [comment]
> It's a bug in the json-languageservice. Fixed [microsoft/vscode-json-languageservice@d3160a7](https://github.com/microsoft/vscode-json-languageservice/commit/d3160a7ac3fda283a4a4964be36f70b6b7627109)

Hi, @aeschli 
so the options here (after fixed@d31607) should be like below? 
```
{
  "comments": "error" // <-- same as allowComments = false, but working
  "trailingCommas": "error" // <-- same as previous behavior, but now configurable
}
```
so the [API documentation](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.languages.json.diagnosticsoptions.html#allowcomments)  is going to be discard `allowComments: false` ?

**Comment by aeschli** [comment]
Ye, I added "comments" and "trailingCommas" to the API: https://github.com/microsoft/monaco-json/blob/c306027c98c84a05885427d9fc8bbc5fd7c21747/monaco.d.ts#L55

`allowComments` is still there for backward compatibility, but is only used when `comments` is not set.
